### PR TITLE
pythonPackages.btrfs: 12 -> 13

### PR DESCRIPTION
**Pull request by Luflosi**
###### Motivation for this change
Version 12 is not available on PyPi, so fetch directly from GitHub instead.
Also install the various python scripts like btrfs-space-calculator.
Test if importing the Python module works fine with `pythonImportsCheck`.
While this library can't be used on macOS to interact with a btrfs filesystem because it is only supported on Linux, btrfs-space-calculator works perfectly fine on macOS. Because of this, I added platforms.darwin to the list of supported platforms.
Also add myself as a maintainer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Poked the upstream about pushing the release to pypi, please hold on.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120638` at 5f7578cf run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>btrfs-heatmap</li>
    <li>python38Packages.btrfs</li>
    <li>python39Packages.btrfs</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/btrfs/default.nix:25:0:
```
   |
25 |     description = "Inspect btrfs filesystems";
   | ^
```
</li>
  </ul>
</details>

**Comment by Luflosi** [comment]
Then I'll convert it to a draft for as long as you haven't gotten a response yet.

**Comment by evils** [comment]
i've got a similar change open at #119753

File: pkgs/development/python-modules/btrfs/default.nix
Comment (evils) [inline pkgs/development/python-modules/btrfs/default.nix:19]: huh, i missed these, doesn't look like the v11 pypi package had these though

File: pkgs/development/python-modules/btrfs/default.nix
Comment (evils) [inline pkgs/development/python-modules/btrfs/default.nix:28]: there's btrfs on darwin?

Comment (evils) [reply pkgs/development/python-modules/btrfs/default.nix:28]: nvm, just read the initial comment in its entirety

**Comment by evils** [comment]
i'm inclined to leave my PR open
if upstream does publish on pypi with the bin stuff installed, you can change this PR to switch back to pypi
or stick with with github and making this an "add executables" PR

as i've not used this since i added this to nixpkgs, i'm glad to see another maintainer added :)

File: pkgs/development/python-modules/btrfs/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/btrfs/default.nix:19]: Can you ask the upstream about setting up entrypoints/scripts?

https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html

File: pkgs/development/python-modules/btrfs/default.nix
Comment (evils) [inline pkgs/development/python-modules/btrfs/default.nix:28]: if `btrfs-space-calculator` is useful on darwin, any reason to think it's not useful on all unix systems?

```suggestion
    platforms = platforms.unix;
```

**Comment by knorrie** [comment]
Hi, thanks for the reminder, @mweinelt.

It seems I also need to fix some text formatting issues. Previously I just included the whole README.md as pypi long description, but that's not a good idea any more since I wrote a lot more stuff in the README. :D 

Also, I'm not a pypi guru, I never use pip myself. So, if someone can give (based on the simple setup.py stuff I have now) some advice on the best way to include the example programs (from bin), so that they're included in the way that's most convenient for you to use, then help is much appreciated.

File: pkgs/development/python-modules/btrfs/default.nix
Comment (Luflosi) [reply pkgs/development/python-modules/btrfs/default.nix:28]: Some python packages don't have `platforms` at all. Is this equivalent to `platforms.unix` or something else?

Comment (evils) [reply pkgs/development/python-modules/btrfs/default.nix:28]: i'm not sure what you're asking
yes, `platforms.unix` is equivalent to `platforms.unix`,
that contains more than linux and darwin

... time elapses ...

turns out buildPythonPackage defaults meta.platforms to that of the interpreter,
which seems to be `platforms.all` for python 2.7 and `linux ++ darwin` for 3.8

the content of the source bin directory all has a python3 shebangs,
so just removing the platforms line is equivalent to your linux ++ darwin line,
and stays correct if our python3 gets built for more platforms
(updating my suggestion to match)

File: pkgs/development/python-modules/btrfs/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/btrfs/default.nix:19]: https://github.com/knorrie/python-btrfs/pull/30

Please fetchpatch and we drop this after the next release.

File: pkgs/development/python-modules/btrfs/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/btrfs/default.nix:28]: > if btrfs-space-calculator is useful on darwin, any reason to think it's not useful on all unix systems?

This is literally the only script that works on non-linux systems. Lets just not go down this rabbit hole and follow the upstreams meta.

https://github.com/knorrie/python-btrfs/blob/113a635793fb565d77fd89ed5e7ecfb8ebfaefc5/setup.py#L35

**Comment by mweinelt** [comment]
The package has no tests, so please add `doCheck = false;` with an appropriate comment.

**Comment by knorrie** [comment]
Ok, thanks, all. I did something and Martin assisted a bit. It's in my 'upstream' develop branch now, and I pushed it to pypi test:

https://test.pypi.org/project/btrfs/
https://github.com/knorrie/python-btrfs/commits/develop

There were some other changes waiting, so this will just be the v13 then.

Let me know if this is what is needed. If so, I'll finish everything and I'm happy to help not slowing you down.

File: pkgs/development/python-modules/btrfs/default.nix
Comment (Luflosi) [reply pkgs/development/python-modules/btrfs/default.nix:19]: I'll wait until your PR is merged, then I'll update this PR.

Comment (mweinelt) [reply pkgs/development/python-modules/btrfs/default.nix:19]: It is already merged and will be part of v13.

File: pkgs/development/python-modules/btrfs/default.nix
Comment (evils) [reply pkgs/development/python-modules/btrfs/default.nix:28]: i'm not sure
on the one hand, if you use it on darwin and are willing to maintain that aspect of the package, it could be fine
on the other hand, even if you conditionally add the other scripts, presumably the python module itself is still useless on darwin, this could cause confusion
running a python script is somewhat trivial, so the cost of not providing it on darwin is small

Comment (SuperSandro2000) [reply pkgs/development/python-modules/btrfs/default.nix:28]:
```suggestion
```

buildPythonPackage defaults to that.

File: pkgs/development/python-modules/btrfs/default.nix
Comment (Luflosi) [reply pkgs/development/python-modules/btrfs/default.nix]: Thank you for your suggestion.

File: pkgs/development/python-modules/btrfs/default.nix
Comment (Luflosi) [reply pkgs/development/python-modules/btrfs/default.nix]: I decided to not change `platforms` for now since btrfs-space-calculator seems to be the only useful thing on darwin, see https://github.com/NixOS/nixpkgs/pull/120638#discussion_r620786877.

Comment (Luflosi) [reply pkgs/development/python-modules/btrfs/default.nix]: I decided to not change `platforms` for now.

**Comment by Luflosi** [comment]
I updated this PR to only include the scripts (see https://github.com/NixOS/nixpkgs/pull/120638#discussion_r620314132) and add myself as maintainer.

**Comment by Luflosi** [comment]
We can update to v13 in a separate PR.

**Review by evils — APPROVED** [review]
thanks

**Comment by knorrie** [comment]
Ok, great, I finished v13.

It's also here already: https://pypi.org/project/btrfs/

**Comment by Luflosi** [comment]
I changed this PR to update this package to version 13.
Why is it preferred to fetch from PyPy instead of from GitHub directly?

**Comment by evils** [comment]
when based on pypi
`nix-shell maintainers/scripts/update.nix --argstr package python3.pkgs.btrfs`
should be able to update the package

(i just confirmed that's the case for the package before my github source change)
(`nix-update python3.pkgs.btrfs` works in either case though)

i'd like to hear if there's other advantages though

**Review by evils — APPROVED** [review]
Result of `nixpkgs-review pr 120638` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>btrfs-heatmap</li>
    <li>python38Packages.btrfs</li>
    <li>python39Packages.btrfs</li>
  </ul>
</details>

**Pull request merged**
